src/univariate/affine.jl

```diff
@@ -0,0 +1,332 @@
+    return mins
+end
+
+function maximum(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmaxes = last.(extreme_matrix)
+    maxes = [σ ⋅ argmax for (σ, argmax) in zip(eachrow(d.σ), eachrow(argmaxes))]
+    return maxes
+end
+
+function extrema(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmins = first.(extreme_matrix)
+    mins = [σ ⋅ argmin for (σ, argmin) in zip(eachrow(d.σ), eachrow(argmins))]
+    argmaxes = last.(extreme_matrix)
+    maxes = [σ ⋅ argmax for (σ, argmax) in zip(eachrow(d.σ), eachrow(argmaxes))]
+    return (mins, maxes)
+end
+
+support(d::AffineDistribution) = affine_support(d.μ, d.σ, support(d.ρ))
+affine_support(μ::Real, σ::Real, support) = μ .+ σ .* support
+
+function affine_support(μ::Real, σ::Real, support::RealInterval) 
+    if σ > 0
+        lower = support.lb
+        upper = support.ub
+    else
+        lower = support.ub
+        upper = support.lb
+    end
+    return RealInterval(μ + σ * lower, μ + σ * upper)
+end
+
+
+#### Conversions
+
+function convert(::Type{AffineDistribution{F,S,Tμ,Tσ,D}}, d::AffineDistribution) where {
+        F<:VariateForm,
+        S<:ValueSupport,
+        D<:Distribution{F, S},
+        Tμ<:ArrayLike,
+        Tσ<:ArrayLike
+    }
+    return AffineDistribution(Tμ(d.μ), Tσ(d.σ), convert(D, d.ρ))
+end
+
+
+function Base.eltype(::Type{<:AffineDistribution{F,S,Tμ,Tσ,D}}) where {
+        F<:VariateForm,
+        S<:ValueSupport,
+        D<:Distribution{F, S},
+        Tμ<:ArrayLike,
+        Tσ<:ArrayLike
+    }
+    T = Core.Compiler.return_type(Base.:*, (Tσ, eltype(D)))
+    return Core.Compiler.return_type(Base.:+, (Tμ, T))
+end
+
+
+#### Parameters
+
+function location(d::AffineDistribution)
+    if hasmethod(location, (typeof(d.ρ),))
+        return d.μ + d.σ * location(d.ρ)
+    else
+        return d.μ
+    end
+end
+function scale(d::AffineDistribution)
+    if hasmethod(scale, (typeof(d.ρ),))
+        return d.σ * scale(d.ρ)
+    else
+        return d.σ
+    end
+end
+shape(d::AffineDistribution) = shape(d.ρ)
+ncategories(d::AffineDistribution) = ncategories(d.ρ)
+params(d::AffineDistribution) = (d.μ, d.σ, d.ρ)
+partype(::AffineDistribution{F,S,Tμ,Tσ,D}) where {F,S,Tμ,Tσ,D} = (Tμ, Tσ)
+
+Base.sign(d::UnivariateAffine) = Base.sign(d.σ)
+
+
+#### Statistics
+
+mean(d::AffineDistribution) = d.μ + d.σ * mean(d.ρ)
+median(d::AffineDistribution) = d.μ + d.σ * median(d.ρ)
+mode(d::AffineDistribution) = d.μ + d.σ * mode(d.ρ)
+modes(d::AffineDistribution) = d.μ .+ d.σ .* modes(d.ρ)
+
+var(d::UnivariateAffine) = d.σ^2 * var(d.ρ)
+std(d::UnivariateAffine) = abs(d.σ) * std(d.ρ)
+skewness(d::UnivariateAffine) = Base.sign(d) * skewness(d.ρ)
+kurtosis(d::UnivariateAffine) = kurtosis(d.ρ)
+
+cov(d::AffineDistribution) = d.σ * cov(d.ρ) * d.σ'
```

Comment:
Probably should be defined only for multivariate distributions, and we should have different implementations based on the type of the scale to reduce allocations.

Reply:
We could do that, but do we want to deviate from `Base`'s behavior (where `cov` returns the variance for univariate outcomes)?